test/graaflib/algorithm/clique/bron_kerbosch_test.cpp

```diff
@@ -0,0 +1,299 @@
+  const auto vertex_4{graph.add_vertex(4)};
+  const auto vertex_5{graph.add_vertex(5)};
+  const auto vertex_6{graph.add_vertex(6)};
+  const auto vertex_7{graph.add_vertex(7)};
+  const auto vertex_8{graph.add_vertex(8)};
+  const auto vertex_9{graph.add_vertex(9)};
+  const auto vertex_10{graph.add_vertex(10)};
+  const auto vertex_11{graph.add_vertex(11)};
+  const auto vertex_12{graph.add_vertex(12)};
+  const auto vertex_13{graph.add_vertex(13)};
+  const auto vertex_14{graph.add_vertex(14)};
+  const auto vertex_15{graph.add_vertex(15)};
+  const auto vertex_16{graph.add_vertex(16)};
+
+  graph.add_edge(vertex_1, vertex_2, 1);
+  graph.add_edge(vertex_1, vertex_15, 1);
+  graph.add_edge(vertex_2, vertex_15, 1);
+
+  graph.add_edge(vertex_2, vertex_4, 1);
+  graph.add_edge(vertex_2, vertex_3, 1);
+  graph.add_edge(vertex_3, vertex_4, 1);
+
+  graph.add_edge(vertex_4, vertex_6, 1);
+  graph.add_edge(vertex_4, vertex_5, 1);
+  graph.add_edge(vertex_5, vertex_6, 1);
+
+  graph.add_edge(vertex_6, vertex_7, 1);
+  graph.add_edge(vertex_6, vertex_8, 1);
+  graph.add_edge(vertex_7, vertex_8, 1);
+
+  graph.add_edge(vertex_8, vertex_9, 1);
+  graph.add_edge(vertex_8, vertex_10, 1);
+  graph.add_edge(vertex_9, vertex_10, 1);
+
+  graph.add_edge(vertex_10, vertex_11, 1);
+  graph.add_edge(vertex_10, vertex_12, 1);
+  graph.add_edge(vertex_11, vertex_12, 1);
+
+  graph.add_edge(vertex_12, vertex_13, 1);
+  graph.add_edge(vertex_12, vertex_14, 1);
+  graph.add_edge(vertex_13, vertex_14, 1);
+
+  graph.add_edge(vertex_14, vertex_15, 1);
+  graph.add_edge(vertex_14, vertex_16, 1);
+  graph.add_edge(vertex_15, vertex_16, 1);
+
+  auto cliques = bron_kerbosch(graph);
+  std::vector<std::vector<vertex_id_t>> expected_cliques{
+      {vertex_1, vertex_2, vertex_15},   {vertex_2, vertex_3, vertex_4},
+      {vertex_5, vertex_4, vertex_6},    {vertex_7, vertex_6, vertex_8},
+      {vertex_8, vertex_9, vertex_10},   {vertex_10, vertex_11, vertex_12},
+      {vertex_12, vertex_13, vertex_14}, {vertex_14, vertex_15, vertex_16},
+  };
+  sort_vectors(cliques, expected_cliques);
+
+  ASSERT_EQ(cliques, expected_cliques);
+}
+
+TYPED_TEST(BronKerbosch, RandomGraphCheckRunTime) {
+  using graph_t = typename TestFixture::graph_t;
+  graph_t graph{};
+
+  std::vector<vertex_id_t> vertices;
+
+  for (int i = 1; i <= 70; ++i) {
+    vertices.push_back(graph.add_vertex(i));
+  }
+
+  std::random_device dev;
+  std::mt19937 rng(dev());
+  std::uniform_int_distribution<std::mt19937::result_type> rand_vertex(0, 69);
+
+  for (int i = 0; i < 1360; ++i) {
+    graph.add_edge(vertices[rand_vertex(rng)], vertices[rand_vertex(rng)], 1);
+  }
+
+  auto start = std::chrono::high_resolution_clock::now();
+  auto cliques = bron_kerbosch(graph);
+  auto stop = std::chrono::high_resolution_clock::now();
+
+  int expected_time = 900;
+
+  ASSERT_TRUE(expected_time >=
+              duration_cast<std::chrono::milliseconds>(stop - start).count());
+}
```

Comment:
Great idea to measure the runtime performance 👍🏻 

Not sure if you saw, but we are also using the google benchmark framework in `/perf`, where we get nice functionalities such as repeated executions of the algorithm after which the average runtime (and memory usage) is reported. There we can also nicely parametrize the benchmark to test for increasingly large graphs.

Such a benchmark might be a better suited place to measure runtime performance of our algorithms than in unit tests. As such I would propose to remove this unit test and to create a follow-up issue in which we can add the benchmark.